src/compiler/checker.ts

```diff
@@ -14091,9 +14194,10 @@ namespace ts {
                 case SyntaxKind.ImportType:
                     return getTypeFromImportTypeNode(<ImportTypeNode>node);
                 // This function assumes that an identifier or qualified name is a type expression
-                // Callers should first ensure this by calling isTypeNode
-                case SyntaxKind.Identifier:
-                case SyntaxKind.QualifiedName:
+                // Callers should first ensure this by calling `isPartOfTypeNode`
+                // TODO(rbuckton): These aren't valid TypeNodes, but we treat them as such because of `isPartOfTypeNode`, which returns `true` for things that aren't `TypeNode`s.

```

Comment:
What's the follow-up needed here?

Reply:
The TODO is to eventually remove these once we fix other related code in `isPartOfTypeNode`, but that was something I considered out-of-scope for this PR. I can update the comment if necessary.